pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -114,13 +114,13 @@ isTlsAllowInsecureConnection_(false) {
     if (clientConfiguration.isUseTls()) {
         using namespace boost::filesystem;
 
-        boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12_client);
+        boost::asio::ssl::context ctx(executor_->io_service_, boost::asio::ssl::context::tlsv1_client);
 
```

Comment:
If there is a macro available to identify boost version, we could use that to pick different version.